stl/inc/ranges

```diff
@@ -763,18 +763,31 @@ namespace ranges {
             { __j - __j } -> convertible_to<_Iota_diff_t<_Ty>>;
         };
 
+    template <class _Wi>
+    struct _Ioterator_category_base {};
+
+    template <incrementable _Wi>
+    struct _Ioterator_category_base<_Wi> {
+        using iterator_category = input_iterator_tag;
+    };
+
     template <weakly_incrementable _Wi>
         requires copyable<_Wi>
-    struct _Ioterator {
+    struct _Ioterator : _Ioterator_category_base<_Wi> {
         // clang-format on
         /* [[no_unique_address]] */ _Wi _Current;
 
-        using iterator_concept  = conditional_t<_Advanceable<_Wi>, random_access_iterator_tag,
+        using iterator_concept = conditional_t<_Advanceable<_Wi>, random_access_iterator_tag,
             conditional_t<_Decrementable<_Wi>, bidirectional_iterator_tag,
                 conditional_t<incrementable<_Wi>, forward_iterator_tag, input_iterator_tag>>>;
-        using iterator_category = input_iterator_tag;
-        using value_type        = _Wi;
-        using difference_type   = _Iota_diff_t<_Wi>;
+        using value_type       = _Wi;
+        using difference_type  = _Iota_diff_t<_Wi>;
+
+        // clang-format off
+        _Ioterator() requires default_initializable<_Wi> = default;
+        // clang-format on
+
+        constexpr _Ioterator(const _Wi& _Value) noexcept(is_nothrow_copy_constructible_v<_Wi>) : _Current(_Value) {}

```

Comment:
We want to avoid defining constructors so this class is ABI-aggregate and can be enregistered. (Yes, I believe that the implicitly-defined constructors do the right thing.)
```suggestion
```

Reply:
Damn that was ugly